### PR TITLE
[mempool] simplify MempoolStatus for interactions with clients

### PR DESCRIPTION
**Pull request by phoenix-antigravity**
preparation for adding JSON-RPC `submit` endpoint
PR introduces MempoolStatus to core libra types (similarly to VMStatus, contains status code and optional message string)
notes:
- PR removes unnecessary mempool-shared-proto crate
- it removes notion of proto types in mempool crate.
  After this PR, AC will communicate with Share Mempool via channel by sending tuples of (MempoolStatus, Option<VmStatus>)
- updates AC to use new types(note: need to double check to what extent it affects existing clients)

### CODE REVIEW

File: types/src/mempool_status.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// Copyright (c) The Libra Core Contributors
```
[the remark this answers is not in the document]
Comment (phoenix-antigravity) [reply types/src/mempool_status.rs]: no regression in behavior though. `vm_error` is defined in similar way
also nice that it eliminates extra crate

**Comment by andll** [comment]
Fixing transient failure in https://github.com/libra/libra/pull/2788

**Pull request closed without merging**